### PR TITLE
Fix Python 3 support

### PR DESCRIPTION
**Pull request by rolando**
This depends on https://github.com/scrapinghub/kafka-scanner/pull/6 and will be rebased once #6 gets merged.

### CODE REVIEW

File: tests/test_kafka_scanner.py [hunk 1]
```diff
@@ -202,7 +226,7 @@ def test_kafka_scan_dedupe_many(self, kafka_consumer_mock):
         self.assertEqual(scanner.scanned_count, 2000)
         self.assertEqual(scanner.dupes_count, 1000)
 
-    def test_kafka_scan_lower_offsets(self, kafka_consumer_mock, batchsize=10000):
+    def _test_kafka_scan_lower_offsets(self, kafka_consumer_mock, batchsize=10000):
```
Comment (kalessin) [inline tests/test_kafka_scanner.py]: which is the point of this kind of change and why to introduce it in a PR for adding Py3 support?

Comment (kalessin) [inline tests/test_kafka_scanner.py]: Suggest style changes/improves/etc in separated PRs than the purpose of the actual one.

Comment (rolando) [inline tests/test_kafka_scanner.py]: It's a change of behavior in how `mock` passes the mocked object to the methods, something related to the additional arguments. With this change those tests fail with the failure below:

```
_______________________________________________________________________________ KafkaScannerTest.test_kafka_scan_partitions _______________________________________________________________________________

self = <test_kafka_scanner.KafkaScannerTest testMethod=test_kafka_scan_partitions>, kafka_consumer_mock = <MagicMock name='KafkaConsumer' spec='KafkaConsumer' id='4419043456'>

    def test_kafka_scan_partitions(self, kafka_consumer_mock):
>       self.test_kafka_scan(num_partitions=3, expected_batches=1)

tests/test_kafka_scanner.py:63:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:1297: in patched
    arg = patching.__enter__()
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:1450: in __enter__
    _name=self.attribute, **kwargs)
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:2368: in create_autospec
    **kwargs)
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:1963: in __init__
    _safe_super(MagicMixin, self).__init__(*args, **kw)
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:1047: in __init__
    _spec_state, _new_name, _new_parent, **kwargs
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:517: in __init__
    self._mock_add_spec(spec, spec_set, _spec_as_instance, _eat_self)
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:574: in _mock_add_spec
    _spec_as_instance, _eat_self)
../../../miniconda3/envs/py35/lib/python3.5/site-packages/mock/mock.py:197: in _get_signature_object
    return func, inspectsignature(sig_func)
../../../miniconda3/envs/py35/lib/python3.5/inspect.py:2989: in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
../../../miniconda3/envs/py35/lib/python3.5/inspect.py:2739: in from_callable
    follow_wrapper_chains=follow_wrapped)
../../../miniconda3/envs/py35/lib/python3.5/inspect.py:2226: in _signature_from_callable
    return _signature_from_function(sigcls, obj)
../../../miniconda3/envs/py35/lib/python3.5/inspect.py:2116: in _signature_from_function
    kind=_VAR_POSITIONAL))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("_name") raised in repr()] Parameter object at 0x107be2318>, name = <MagicMock name='KafkaConsumer._message_generator.__code__.co_varnames.__getitem__()' id='4424740936'>
kind = <_ParameterKind.VAR_POSITIONAL: 2>

    def __init__(self, name, kind, *, default=_empty, annotation=_empty):

        if kind not in (_POSITIONAL_ONLY, _POSITIONAL_OR_KEYWORD,
                        _VAR_POSITIONAL, _KEYWORD_ONLY, _VAR_KEYWORD):
            raise ValueError("invalid value for 'Parameter.kind' attribute")
        self._kind = kind

        if default is not _empty:
            if kind in (_VAR_POSITIONAL, _VAR_KEYWORD):
                msg = '{} parameters cannot have default values'.format(kind)
                raise ValueError(msg)
        self._default = default
        self._annotation = annotation

        if name is _empty:
            raise ValueError('name is a required attribute for Parameter')

        if not isinstance(name, str):
>           raise TypeError("name must be a str, not a {!r}".format(name))
E           TypeError: name must be a str, not a <MagicMock name='KafkaConsumer._message_generator.__code__.co_varnames.__getitem__()' id='4424740936'>

../../../miniconda3/envs/py35/lib/python3.5/inspect.py:2417: TypeError
======================================================================== 13 failed, 19 passed, 1 pytest-warnings in 15.57 seconds =========================================================================
```

**Comment by rolando** [comment]
@kalessin I've moved the travis stuff to its own PR: https://github.com/scrapinghub/kafka-scanner/pull/6

**Comment by rolando** [comment]
I have restricted this PR to py3-only fixes. Here is the build log: https://travis-ci.org/rolando-contrib/kafka-scanner/builds/214366149

**Comment by kalessin** [comment]
@rolando I added a much more simple .travis configuration file (I don't know why you added a so big one :P even with installations that we don't need at all)

can recreate this PR only with the changes required for python3 support?

**Comment by rolando** [comment]
OK, that works for me. I've updated the PR.

**Comment by kalessin** [comment]
merged! thanks @rolando

**Comment by rolando** [comment]
Thank you for looking at this!

**Pull request merged**